pkg/controllers/kube-apiserver.go

```diff
@@ -221,27 +223,34 @@ func (s *KubeAPIServer) configureOAuth(cfg *config.MicroshiftConfig) error {
 func (s *KubeAPIServer) Run(ctx context.Context, ready chan<- struct{}, stopped chan<- struct{}) error {
 	defer close(stopped)
 	errorChannel := make(chan error, 1)
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
 
 	// run readiness check
 	go func() {
-		restConfig, err := clientcmd.BuildConfigFromFlags("", s.kubeconfig)
+		err := wait.PollImmediateWithContext(ctx, time.Second, kubeAPIStartupTimeout*time.Second, func(ctx context.Context) (bool, error) {
+			restConfig, err := clientcmd.BuildConfigFromFlags("", s.kubeconfig)
+			if err != nil {
+				return false, err
+			}
+			restConfig.NegotiatedSerializer = serializer.NewCodecFactory(runtime.NewScheme())
+
+			restClient, err := rest.UnversionedRESTClientFor(restConfig)
+			if err != nil {
+				return false, err
+			}
+
+			var status int
+			if err := restClient.Get().AbsPath("/readyz").Do(ctx).StatusCode(&status).Error(); err != nil {
+				return false, nil
```

Comment:
Would it make sense to log a "waiting" message here?

Reply:
Sure, done.